sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1067,7 +1072,16 @@ static ipv6_addr_t *_src_addr_selection(gnrc_netif_t *netif,
         }
         else if (candidate_scope > dst_scope) {
             DEBUG("winner for rule 2 (larger scope) found\n");
-            winner_set[i] += candidate_scope;
```

Comment:
The only issue is the case `source(SA) < source(SB) < source(D)`, I find sources that say that in that case they both have the same preference and others that sat `SB` should be preferred. The `RFC` makes me think it should be `SA`, but with the `if (candidate_scope > dst_scope)` they will both get the same preference.

```
if (A < B) {
    if( A < D) {
        choose B
    } else {
        choose A
    }
}
else {
    choose A
}
```


BTW: its been a while since something so simple has given me sucha  headache ahahaha, just 2 ifs

Reply:
> BTW: its been a while since something so simple has given me sucha headache ahahaha, just 2 ifs

Yeah... but the wording in the RFC could be a little bit more trivial IMHO so no wonder it gives a headache :sweat_smile:

Reply:
Actually I agree with you: the RFC is pretty clear on the case you talked about: It should be `SB`: `if Scope(SA) < Scope(SB)` (true in our case) then `If Scope(SA) < Scope(D),` (also true in our case) pick `then prefer SB […]`.

Reply:
(so we need an else case where `SB` is picked if `Scope(SA) < Scope(SB) < Scope(D)`, but `SC`, if `Scope(SA) < Scope(SB) < Scope(D) < Scope(SC)`... I think that makes the scoring a bit more complicated ^^"

Reply:
@fjmolinas I think my latest proposal should make the best selection.